sycl/test/basic_tests/vec_convert_f_to_f.cpp

```diff
@@ -0,0 +1,110 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
+// RUN: env SYCL_DEVICE_TYPE=CPU %t.out
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+// RUN: %ACC_RUN_PLACEHOLDER %t.out
+//==------------ vec_convert.cpp - SYCL vec class convert method test ------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include <CL/sycl.hpp>
+
+#include <cassert>
+
+using namespace cl::sycl;
+
+template <typename T, typename convertT, int roundingMode>
+class kernel_name;
+
+template <int N>
+struct helper;
+
+template <>
+struct helper<0> {
+  template <typename T, int NumElements>
+  static void compare(const vec<T, NumElements> &x,
+                      const vec<T, NumElements> &y) {
+    const T xs = x.template swizzle<0>();
+    const T ys = y.template swizzle<0>();
+    assert(xs == ys);
+  }
+};
+
+template <int N>
+struct helper {
+  template <typename T, int NumElements>
+  static void compare(const vec<T, NumElements> &x,
+                      const vec<T, NumElements> &y) {
+    const T xs = x.template swizzle<N>();
+    const T ys = y.template swizzle<N>();
+    helper<N - 1>::compare(x, y);
+    assert(xs == ys);
+  }
+};
+
```

Comment:
Please outline duplicated code into a header file with helpers

Reply:
There should I place the header?